sort.c

```diff
@@ -10,30 +10,52 @@
 
 #include "redisraft.h"
 
-static void replySortedArray(RedisModuleCtx *ctx, RedisModuleCallReply * reply)
+static void replySortedArray(RedisModuleCtx *ctx, RedisModuleCallReply * reply, int reply_type)
 {
     size_t len = RedisModule_CallReplyLength(reply);
-    if (len == 0) {
-        RedisModule_ReplyWithNullArray(ctx);
-        return;
-    }
 
-    RedisModuleDict * dict = RedisModule_CreateDict(ctx);
-    for(int i=0; i < len; i++) {
-        RedisModuleCallReply * entry = RedisModule_CallReplyArrayElement(reply, i);
+    RedisModuleDict *dict = RedisModule_CreateDict(ctx);
+    for(size_t i = 0; i < len; i++) {
+        long *val;
+        RedisModuleCallReply * entry;
+        switch (reply_type) {
+            case REDISMODULE_REPLY_ARRAY:
+                entry = RedisModule_CallReplyArrayElement(reply, i);
+                break;
+            case REDISMODULE_REPLY_SET:
+                entry = RedisModule_CallReplySetElement(reply, i);
+                break;
+        }
+
         size_t entry_len;
         const char * entry_str = RedisModule_CallReplyStringPtr(entry, &entry_len);
-        RedisModule_DictSetC(dict, (char *) entry_str, entry_len, NULL);
+        val = RedisModule_DictGetC(dict, (char *) entry_str, entry_len, NULL);
+        if (!val) {
+            val = RedisModule_Calloc(1, sizeof(long));
+        }
+        (*val)++;
+        RedisModule_DictSetC(dict, (char *) entry_str, entry_len, val);
     }
 
-    RedisModule_ReplyWithArray(ctx, len);
+    switch (reply_type)
+    {
+        case REDISMODULE_REPLY_ARRAY:
+            RedisModule_ReplyWithArray(ctx, len);
+            break;
+        case REDISMODULE_REPLY_SET:
+            RedisModule_ReplyWithSet(ctx, len);
+            break;
```

Comment:
Should probably pass the return value from RedisModule_ReplyWith...

Reply:
you mean check?

Reply:
I think `replySortedArray` should bail out if a `ReplyWith...` returns an error, and itself return an error for callers to be able to handle it.

Reply:
so looking at the docs, it says

```
RedisModule_ReplyWithArray

int RedisModule_ReplyWithArray(RedisModuleCtx *ctx, long len);

Reply with an array type of 'len' elements. However 'len' other calls to ReplyWith* style functions must follow in order to emit the elements of the array.

When producing arrays with a number of element that is not known beforehand the function can be called with the special count REDISMODULE_POSTPONED_LEN, and the actual number of elements can be later set with RedisModule_ReplySetArrayLength() (which will set the latest "open" count if there are multiple ones).

The function always returns REDISMODULE_OK.
```

is the point to be protective for the future where it could fail?